core/CMakeLists.txt

```diff
@@ -336,6 +336,10 @@ if (X86 AND X64)
   set(ARCH_SRCS ${ARCH_SRCS} arch/${ARCH_NAME}/x86_to_x64.c)
 endif (X86 AND X64)
 
+if (AARCHXX)
+  set(ARCH_SRCS ${ARCH_SRCS} arch/${ARCH_NAME}/asm_offsets.cpp)
```

Comment:
We don't want to mix C++ code into core DR.  libdynamorio.so cannot rely on a C++ library.  Even if this .cpp never pulls in any runtime deps: it feels risky and fragile, inviting new devs to put in unsupported C++ code thinking it's normal.

Reply:
Yes, that makes sense. But I can't put my `static_assert` code into a non-core part of DR because it needs access to structs that are not exported. So should I go back to using `char x1[condition ? 1 : -1]` instead of `static_assert`?

Reply:
Please see my original comment https://github.com/DynamoRIO/dynamorio/pull/7202#discussion_r1924259563 which suggests using _Static_assert in C and static_assert in C++ behind a macro unifying them.

Reply:
The checks are only happening in one place, currently `asm_offsets.cpp`, though it sounds like I should go back to using `asm_offsets.c`, as in the first version of this PR. I don't see any need to perform the checks in two languages.